pytorch_forecasting/data/timeseries/_timeseries.py

```diff
@@ -2342,6 +2347,51 @@ def __getitem__(self, idx: int) -> tuple[dict[str, torch.Tensor], torch.Tensor]:
             (target, weight),
         )
 
+    def __precompute__(self, batch_size, shuffle, drop_last):
+        """
+        Precompute sample for model
+
+        Args:
+        batch_size : int, optional, default=64
+            batch size for training model. Defaults to 64.
+        shuffle : bool
+            indicate whether to shuffle the data
+        drop_last : bool
+            indicate whether to drop last
+        """
+        sampler = TimeSynchronizedBatchSampler(
+            SequentialSampler(self),
+            batch_size=batch_size,
+            shuffle=shuffle,
+            drop_last=drop_last,
+        )
+
+        for batch in sampler:
+            for idx in batch:
+                batch_result = self.__item_tensor__(idx)
+                self.precompute_cache.append(batch_result)
+
+    def __getitem__(self, idx: int) -> tuple[dict[str, torch.Tensor], torch.Tensor]:
+        """
+        Get sample for model
+
+        Args:
+            idx (int): index of prediction (between ``0`` and ``len(dataset) - 1``)
+
+        Returns:
+            tuple[dict[str, torch.Tensor], torch.Tensor]: x and y for model
+        """
+        if self.precompute:
+            if self.precompute_idx >= len(self.precompute_cache):
```

Comment:
could we move this logic into an extra private function `_retrieve_precomputed_and_increment_idx` (or similar)? Just to make this function easier to read, and clearer what is happening.